c_api/tests/c_api_behavior_test.cpp

```diff
@@ -117,39 +117,51 @@ TEST_F(CapiTest, TestBehaviorGetRuntimeCapabilities) {
 
 TEST_F(CapiTest, TestBehaviorAllocateMemory) {
   auto inner = [&](TiArch arch) {
-    if (ti::is_arch_available(arch)) {
-      // Attempt to allocate memory with size of 1024
-      TiRuntime runtime = ti_create_runtime(arch);
-      for (int i = 0; i < 4; ++i) {
-        TiMemoryAllocateInfo allocate_info;
-        allocate_info.size = 1024;
-        allocate_info.usage = TI_MEMORY_USAGE_STORAGE_BIT << i;
-        TiMemory memory = ti_allocate_memory(runtime, &allocate_info);
-        TI_ASSERT(memory != TI_NULL_HANDLE);
-        ti_free_memory(runtime, memory);
-      }
+    if (!ti::is_arch_available(arch)) {
+      TI_WARN("arch {} is not supported, so this test is skipped", arch);
+    }
 
-      // runtime and allocate_info are both null
-      {
-        ti_allocate_memory(TI_NULL_HANDLE, nullptr);
-        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
-      }
+    // Attempt to allocate memory with size of 1024
+    TiRuntime runtime = ti_create_runtime(arch);
```

Comment:
nit: better split these to 5 smaller / separate tests to keep tests minimal and isolated ;)